lib/app_profiler/server.rb

```diff
@@ -0,0 +1,292 @@
+# frozen_string_literal: true
+
+require "socket"
+require "rack"
+require "tempfile"
+
+# This module provides a means to start a golang-inspired profile server
+# it is implemented using stdlib and Rack to avoid additional dependencies
+
+module AppProfiler
+  module Server
+    HTTP_OK = 200
+    HTTP_BAD_REQUEST = 400
+    HTTP_NOT_FOUND = 404
+    HTTP_NOT_ALLOWED = 405
+    HTTP_CONFLICT = 409
+
+    TRANSPORT_UNIX = "unix"
+    TRANSPORT_TCP = "tcp"
+
+    DEFAULTS = {
+      enabled: false,
+      transport: TRANSPORT_UNIX,
+      cors: true,
+      cors_host: "*",
+      port: 0,
+      duration: 30,
+    }
+
+    mattr_accessor :enabled, default: DEFAULTS[:enabled]
+    mattr_accessor :transport, default: DEFAULTS[:transport]
+    mattr_accessor :cors, default: DEFAULTS[:cors]
+    mattr_accessor :cors_host, default: DEFAULTS[:cors_host]
+    mattr_accessor :port, default: DEFAULTS[:port]
+    mattr_accessor :duration, default: DEFAULTS[:duration]
+
+    class ProfileApplication
```

Comment:
Do we want children of the server module to be public or private API? Perhaps we should use `private_constant` to make some of this private.